horreum-backend/src/main/resources/db/changeLog.xml

```diff
@@ -4665,4 +4665,26 @@
         <dropColumn tableName="schema" columnName="token" />
         <dropTable tableName="test_token" />
     </changeSet>
+    <changeSet id="125" author="lampajr">
+        <validCheckSum>ANY</validCheckSum>
+        <sql>
+            -- still need db procedure until https://hibernate.atlassian.net/browse/HHH-17314 is fixed in quarkus
+            CREATE OR REPLACE PROCEDURE calc_dataset_view(datasetId bigint) AS $$
+            BEGIN
+            WITH view_agg AS (
+                SELECT
+                    vc.view_id, vc.id as vcid, array_agg(DISTINCT label.id) as label_ids, jsonb_object_agg(label.name, lv.value) as value FROM dataset_schemas ds
+                JOIN label ON label.schema_id = ds.schema_id
+                JOIN viewcomponent vc ON vc.labels ? label.name
+                JOIN label_values lv ON lv.label_id = label.id AND lv.dataset_id = ds.dataset_id
```

Comment:
this is the only change: adding `lv.dataset_id = ds.dataset_id` in the join condition